File: APP03022-mrch.backend.somx.catalogos-api/src/main/java/com/sodimac/catman/api/exception/ConflictException.java
package com.sodimac.catman.api.exception;

public class ConflictException extends RuntimeException {
    
    private final String errorType;
    
    public ConflictException(String message) {
        super(message);
        this.errorType = "ConflictError";
    }
    
    public ConflictException(String errorType, String message) {
        super(message);
        this.errorType = errorType;
    }
    
    public String getErrorType() {
        return errorType;
    }
}

File: APP03022-mrch.backend.somx.util-api/src/exceptions/ConflictException.ts
export class ConflictException extends Error {
    public readonly errorType: string;

    constructor(message: string, errorType: string = 'ConflictError') {
        super(message);
        this.name = 'ConflictException';
        this.errorType = errorType;

        if (Error.captureStackTrace) {
            Error.captureStackTrace(this, ConflictException);
        }
    }
}

export default ConflictException;

